conbench/entities/history.py

```diff
@@ -302,30 +306,40 @@ def set_z_scores(benchmark_results: List[BenchmarkResult], baseline_commit: Comm
         )
 
 
-def _query_distribution_stats_by_run_id(
-    run_id: str,
+def _query_and_calculate_distribution_stats(
+    contender_run_id: str,
     baseline_commit: Commit,
     case_id: Optional[str],
     context_id: Optional[str],
```

Comment:
Note from sync meeting: case_id and context_id are where we narrow this analysis down to "just matching benchmark results in the contender_run".

Reply:
Another note: for code readability, it would probably make sense to pass along the `contender_benchmark_results` instead of `contender_run_id`. But "filtering the distribution query to a potentially-large list of case/context pairs" seems hard to do in SQLAlchemy. That's why I didn't do it before.

What if we didn't filter case/context at all in the distribution query, and instead did it in pandas after execution? That could probably speed up the query since we'd remove the complicated join!